UnityProject/Assets/Scripts/Machines/Microwave.cs

```diff
@@ -124,30 +124,50 @@ public void RpcStartCooking()
 	private void FinishCooking()
 	{
 		spriteRenderer.sprite = SPRITE_OFF;
-		if (isServer)
+
+		if (isServer && mealCount > 0)
 		{
 			GameObject mealPrefab = CraftingManager.Meals.FindOutputMeal(meal);
-			SpawnResult result = Spawn.ServerPrefab(mealPrefab, GetComponent<RegisterTile>().WorldPosition, transform.parent);
+			Vector3Int spawnPosition = GetComponent<RegisterTile>().WorldPosition;
 
-			//If the resulting meal has a stackable component, set the amount to mealCount to ensure that food in = food out.
+			SpawnResult result = Spawn.ServerPrefab(mealPrefab, spawnPosition, transform.parent);
 			Stackable stck = result.GameObject.GetComponent<Stackable>();
 
-			if (stck != null && mealCount != 0)
+			if (stck != null)   // If the meal has a stackable component, set the correct number of meals
 			{
-				//Get difference between new item's initial amount and the amount held by mealCount (amount of ingredient).
-				int stckChanger = mealCount-stck.Amount;
+				int spawnedSoFar = stck.InitialAmount;
 
-				//If stckChanger is 0, do nothing.
-				//If stckChanger is positive, add to stack.
-				if (stckChanger > 0)
+				while (spawnedSoFar < mealCount)
 				{
-					stck.ServerIncrease(stckChanger);
-				} else if (stckChanger < 0)
+					int spaceInStack = stck.MaxAmount - stck.Amount;
+
+					if (spaceInStack > 0)
+					{
+						// Add all the missing meals to the stack, or as many as the stack can contain
+						int mealDeficit = mealCount - spawnedSoFar;
+						int stacksToAdd = Mathf.Min(spaceInStack, mealDeficit);
+						stck.ServerIncrease(stacksToAdd);
```

Comment:
Maybe out of scope of this but it seems like this must be a pretty common problem.

A nice solution might be to have this ServerIncrease return the extra amount it couldn't spawn due to exceeding the stack - you could then just loop calling the method and spawning a new stack each time until it returned 0.

Reply:
I'll have a look at adjusting Stackable to make it a bit easier to work with